src/test/groovy/net/rubygrapefruit/platform/file/OsxFileEventsTest.groovy

```diff
@@ -23,149 +23,164 @@ import org.junit.Rule
 import org.junit.rules.TemporaryFolder
 import spock.lang.Requires
 import spock.lang.Specification
-import spock.lang.Timeout
+import spock.util.concurrent.AsyncConditions
 
 @Requires({ Platform.current().macOs })
-@Timeout(20)
 class OsxFileEventsTest extends Specification {
     @Rule
     TemporaryFolder tmpDir
     final OsxFileEventFunctions fileEvents = Native.get(OsxFileEventFunctions.class)
     FileWatcher watcher
+    def callback = new TestCallback()
+    File dir
+
+    def setup() {
+        dir = tmpDir.newFolder()
+    }
+
+    def cleanup() {
+        stopWatcher()
+    }
 
     def "caches file events instance"() {
         expect:
         Native.get(OsxFileEventFunctions.class) is fileEvents
     }
 
     def "can open and close watcher on a directory without receiving any events"() {
-        def changes = startWatcher()
-
-        expect:
-        changes.empty
+        when:
+        startWatcher(dir)
 
-        cleanup:
-        stopWatcher()
+        then:
+        0 * _
     }
 
     def "can open and close watcher on a directory receiving an event"() {
         given:
-        def dir = tmpDir.newFolder()
-        def changes = startWatcher(dir.absolutePath)
+        startWatcher(dir)
 
         when:
+        def expectedChanges = expectThat pathChanged(dir)
         new File(dir, "a.txt").createNewFile()
-        waitForFileSystem()
 
         then:
-        changes == [dir.canonicalPath + "/"]
-
-        cleanup:
-        stopWatcher()
+        expectedChanges.await()
     }
 
     def "can open and close watcher on a directory receiving multiple events"() {
-        def latency = 0.5
         given:
-        def dir = tmpDir.newFolder()
-        def changes = startWatcher(latency, dir.absolutePath)
+        def latency = 0.3
+        startWatcher(latency, dir)
 
         when:
+        def expectedChanges = expectThat pathChanged(dir)
         new File(dir, "a.txt").createNewFile()
-        waitForFileSystem()
 
         then:
-        changes == [dir.canonicalPath + "/"]
+        expectedChanges.await()
 
-        when: "directory changed almost immediately again"
-        new File(dir, "b.txt").createNewFile()
-        waitForFileSystem()
-
-        then: "change is not reported"
-        changes == [dir.canonicalPath + "/"]
-
-        when: "directory is changed after latency period"
+        when:
+        expectedChanges = expectThat pathChanged(dir)
         Thread.sleep((long) (latency * 1000 + 100))
```

Comment:
Does it need this considering .await() is invoked on line 88?

Reply:
Yes, because macOS won't report another change until `latency` is elapsed.